overlord/schema/manager.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package schema
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/overlord/state"
+)
+
+type State interface {
+	DB() *sql.DB
```

Comment:
Is this actually needed? You might be able to simply move the `TxnRunner` methods here and only access the DB via `WithinTxn`

Reply:
Nope, gone.